include/boost/spirit/home/x3/core/skip_over.hpp

```diff
@@ -91,9 +151,9 @@ namespace boost { namespace spirit { namespace x3
 
     template <typename Iterator, typename Context>
     inline void skip_over(
```

Comment:
I guess throwing expectation failure inside a skipper and catching it in a rule does work but I don't think there any use for it. That's probably an omission that it's not catched here. Is it really need to propagate expectation failures outside from skippers?

Reply:
Consider the case like below:

<https://github.com/boostorg/spirit/pull/788/files#diff-098440cf78f4367f618d21a85375e53988bc19c9d193a26876e4fb40d81ff5f1R448-R451>

```cpp
TEST_FAILURE("a..b.z", lit('a') >> 'b', lit('.') > '.', {
    BOOST_TEST(x.which() == "'.'"sv);
    BOOST_TEST(x.where() == "z"sv);
});
```

Traditionally in throwing mode, this kind of grammar led to an expectation_failure being thrown. To get the same `x.where()` and `x.where()` values in non-throwing mode, we need to propagate the failures.